src/repl/interpreter.rs

```diff
@@ -171,15 +171,68 @@ impl ClarityInterpreter {
         Ok(result)
     }
 
+    pub fn run_ast(
```

Comment:
Something that would be useful for both the LSP and Orchestra, would be the ability to decouple the analysis from the execution. At the moment, the contract is being saved in the AnalysisDatabase after being executed (which is an unnecessary step that we could avoid in these 2 products), so this would need to change. 
I recently introduced the method
https://github.com/hirosystems/clarity-repl/blob/d3eaac63ee32d0d07aad33ec5ea9315122c9f535/src/repl/interpreter.rs#L342
that could help.

Reply:
Thanks for the reminder about that. Is your suggestion that we can just remove this new `run_ast` method, and instead use `save_contract` and `execute` in its place?

Reply:
Apologies for the latency, let's disregard my comment!